sql/moz-fx-data-shared-prod/fenix_derived/google_ads_campaign_cost_breakdowns_v1/query.sql

```diff
@@ -0,0 +1,79 @@
+WITH campaigns AS (
+  SELECT DISTINCT
+    id,
+    name,
+    FORMAT("%s (%s)", name, CAST(id AS string)) AS fenix_compatible_campaign_name
+  FROM
+    `moz-fx-data-bq-fivetran`.google_ads.campaign_history
+),
+install_dou_metrics AS (
+  SELECT
+    fenix_marketing_metrics.adjust_campaign AS fenix_marketing_metrics_adjust_campaign,
+    fenix_marketing_metrics.submission_date AS date,
+    COALESCE(SUM(fenix_marketing_metrics.new_profiles), 0) AS new_profiles_sum,
+    COALESCE(SUM(fenix_marketing_metrics.dau), 0) AS dau_sum
+  FROM
+    `moz-fx-data-shared-prod.fenix.marketing_attributable_metrics` AS fenix_marketing_metrics
+  WHERE
+    adjust_network = "Google Ads ACI"
+  GROUP BY
+    fenix_marketing_metrics_adjust_campaign,
+    date
+)
+SELECT
+  campaigns.name AS campaign_name,
+  stats.date AS date,
+  -- Total spend per-campaign
+  stats.cost_micros AS campaign_spend_in_micros,
+  -- Impressions and clicks over time for each campaign (clicks of our ads)
+  impressions AS ad_impressions,
+  install_dou_metrics.new_profiles_sum AS installs,
+  install_dou_metrics.dau_sum AS dous,
+  conversions as ad_conversions,
+  clicks AS revenue_generating_ad_clicks,
+  -- Cost per-install for each campaign ($/new profiles)
+  CASE
+  WHEN
+    install_dou_metrics.new_profiles_sum = 0
+  THEN
+    0
+  ELSE
+    stats.cost_micros / install_dou_metrics.new_profiles_sum
+  END
+  AS cost_per_install_micros,
+  -- Cost per-DOU for each campaign (microunits of local currency/DOU)
+  CASE
+  WHEN
+    install_dou_metrics.dau_sum = 0
+  THEN
+    0
+  ELSE
+    stats.cost_micros / install_dou_metrics.dau_sum
+  END
+  AS cost_per_dou_micros,
+  -- Cost per-Ad Click for each campaign (microunits of local currency/Ad Clicks)
+  CASE
+  WHEN
+    clicks = 0
+  THEN
+    0
+  ELSE
+    stats.cost_micros / clicks
+  END
+  AS cost_per_click_micros
+FROM
+  `moz-fx-data-bq-fivetran`.google_ads.campaign_stats AS stats
+JOIN
+  campaigns
+USING
+  (id)
+JOIN
+  install_dou_metrics
+ON
+  (stats.date = install_dou_metrics.date)
+  AND (
+    campaigns.fenix_compatible_campaign_name = install_dou_metrics.fenix_marketing_metrics_adjust_campaign
+  )
```

Comment:
Optional for future reference: sometimes it's nice to name these columns the same in the CTE so you can use `USING`:
```
JOIN
  install_dou_metrics
USING (date, campaign_name)
```

Reply:
Hundo P. The thing I ran into here was that I wanted to _clearly_ document that this is not actually the campaign_name: it's a bastardization used by fenix, that concatenates the actual campaign_name to a numeric id and then _calls_ it campaign_name. This took me a non-negligible amount of time to figure out precisely because things are named for convenience rather than accuracy in several places, and I don't want the next DE to have to go through that detective work as well.

Reply:
Fair point, we'll leave it as-is :) thank's for being thoughtful to future us!